services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -659,6 +664,22 @@ protected void onPartitionStatusChange(OfflinePushStatus offlinePushStatus) {
     checkWhetherToStartBufferReplayForHybrid(offlinePushStatus);
   }
 
+  private boolean disableReplica(PartitionAssignment partitionAssignment, Optional<String> leaderErrorInfo) {
+    String errorInfo = leaderErrorInfo.orElse("");
+    if (errorInfo.isEmpty()) {
+      return false;
+    }
+    String[] tokens = errorInfo.split("_", 2);
+    String resourceName = partitionAssignment.getTopic();
+    helixAdminClient.enablePartition(
+        false,
+        clusterName,
+        tokens[0],
+        resourceName,
+        Collections.singletonList(HelixUtils.getPartitionName(resourceName, Integer.parseInt(tokens[1]))));
+    return true;
+  }
+
   @Override
   public void onExternalViewChange(PartitionAssignment partitionAssignment) {
```

Comment:
There are two triggers of push status change:
1. EV change.
2. Partition status change.
I think we need to handle both.

If we could handle the leader error inside `PushStatusDecider` by a callback, it will cover both.

Reply:
Added a callback to reset the replicas from `PushStatusDecider`